.clang-format

```diff
@@ -0,0 +1,96 @@
+---
+Language:        Cpp
+# BasedOnStyle:  LLVM
+AccessModifierOffset: -2
+AlignAfterOpenBracket: Align
```

Comment:
Kodi code guidelines say to never align vertically

Reply:
for method wrapping I don't believe we have any guidelines so this would be a new rule. Personally I think it's nice but I haven't spent much time tinkering with the various settings.

Reply:
Depends on whether you think the reasoning ("becomes problematic on updates to see the change, if the whitespace need to change on all") applies here

Reply:
yeah that is an issue but as we're not consistent today any setting we pick will be an issue. I would prefer to settle on a format, run it on everything at once and then continue on as normal to make sure we only pay that cost once in a single non-functional cosmetic only PR

Reply:
We'll have to go through a one time cleanup anyway it seems. All we could do is at least do it in small chunks and not one big lump. As long as we agree on what standard

Reply:
that sounds good to me.

Reply:
Spending some time thinking about it now when updating the clang-format with the suggestions I think this setting makes sense. It gives a certain flow to the code which makes it quite easy to quickly scan over as argument blocks gets pushed more towards the right column of the screen.